packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -453,6 +461,56 @@ abstract contract TimelockGuard is IGuard {
         _checkCombinedConfig(callingSafe);
     }
 
+    /// @notice Clears the timelock guard configuration for a Safe.
+    /// @dev Note: Clearing the configuration also cancels all pending transactions.
+    ///      This function is intended for use when a Safe wants to permanently remove
+    ///      the TimelockGuard configuration. Typical usage pattern:
+    ///      1. Safe disables the guard via GuardManager.setGuard(address(0)).
+    ///      2. Safe calls this clearTimelockGuard() function to remove stored configuration.
+    ///      3. If Safe later re-enables the guard, it must call configureTimelockGuard() again.
+    function clearTimelockGuard() external {
```

Comment:
optional nit to be more explicit, from the function name I thought this would also remove the guard from storage, even though the natspec says otherwise
```suggestion
    function clearTimelockGuardConfig() external {
```